File: May-LeetCoding-Challenge/03-Ransom-Note/Ransom-Note.cs
public class Solution {
    public bool CanConstruct(string ransomNote, string magazine) {
        
        Dictionary<char, int> dicMagazine =  new Dictionary<char, int>();
        foreach (char c in magazine)
        {
           if(dicMagazine.ContainsKey(c))
           {
              dicMagazine[c] ++;
           }
            else
            {
                dicMagazine.Add(c,1);
            }    
        }
        foreach (char r in ransomNote)
        {
            if (dicMagazine.ContainsKey(r))
            {
               dicMagazine[r] --;
                if ( dicMagazine[r] == 0)
                {
                    dicMagazine.Remove(r);
                }
            }
            else
            {
                return false;
            }
         
        } 
           return true;
    }
}

File: May-LeetCoding-Challenge/03-Ransom-Note/Ransom-Note.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        vector<int> nums(26, 0);
        for (auto &c: magazine) nums[c - 'a']++;
        for (auto &c: ransomNote) 
            if (nums[c - 'a']-- <= 0) return false;
        return true;
    }
};